authors/apps/article/views.py

```diff
@@ -142,8 +155,125 @@ def post(self, request, **kwargs):
+            like.is_disliked = False
+            like.is_liked = True
+            like.save()
+            return Response(
+                {"response": "article successfully liked"})
+        except Exception:
+            LikeDislikeArticle(
+                liker=liker,
+                article=article,
+                is_liked=True).save()
+            return Response(
+                {"response": "article successfully liked"})
+
+    def get(self, request, **kwargs):
+        """
+        Return likes of an article
+        """
+
+        likes = None
+        slug = kwargs.get('slug')
+        try:
+            article = Article.objects.get(slug=slug)
+        except Exception:
+            return Response({"response": no_article},
+                            status=status.HTTP_204_NO_CONTENT)
+        likes = LikeDislikeArticle.objects.filter(article=article)
+        return Response({"response": len(
+            [like for like in likes if like.is_liked])})
+
+
+class Dislike(APIView):
+    """
+    Dislike views
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, **kwargs):
+        """
+        Return dislikes of an article
+        """
+
+        try:
+            disliked_article = get_article(kwargs)
+        except Exception:
+            return Response({"response": "No article found"},
+                            status=status.HTTP_204_NO_CONTENT)
+        dislikes = LikeDislikeArticle.objects.filter(
+            article=disliked_article)
+        dislikes_number = [
+            dislike for dislike in dislikes if dislike.is_disliked]
+        return Response({"response": len(dislikes_number)})
+
+    def post(self, request, **kwargs):
+        """
+        Like an article.
+        """
+        disliked_article = None
+        disliker = None
+        try:
+            disliked_article = get_article(kwargs)
+            disliker = request.user
+        except Exception:
+            return Response({"response": "No article found"},
+                            status=status.HTTP_204_NO_CONTENT)
+
+        try:
+            dislike = LikeDislikeArticle.objects.get(
+                article=disliked_article,
+                liker=disliker)
+            if dislike.is_disliked:
+                dislike.is_disliked = False
+                dislike.save()
+                return Response(
+                    {"response": undisliked_response})
+            dislike.is_liked = False
+            dislike.is_disliked = True
+            dislike.save()
+            return Response(
+                {"response": disliked_article_response},
+                status=status.HTTP_200_OK)
+        except Exception as e:
+            LikeDislikeArticle(
+                liker=disliker,
+                article=disliked_article,
+                is_disliked=True).save()
+            return Response(
+                {"response": "article successfully disliked"})
```

Comment:
The limit is 250 lines of code per file, why do we have 279 here?

Reply:
The reason I haven't included the http status code is because of the 250 of limit line of code @evamaina 
@code-sleuth the line of code is 250 but the blank space makes it 279